### PR TITLE
Bug fix for GOES sensorAzimuthAngle

### LINKED ISSUE #1229: [Bug]  GOES converter computes incorrect sensorAzimuthAngle
## Current behavior (describe the bug)

The GOES converter followed a [NOAA web page](https://www.ncdc.noaa.gov/gridsat/docs/Angle_Calculations.pdf) for computing satellite azimuth angle, but it is incorrect.

## To Reproduce

A graphic of the output produced by the wrong code is shown here.
g16_8km_MetaData_sensorAzimuthAngle

## Expected behavior

A corresponding [Australian BOM page](https://opus.nci.org.au/display/NDP/Himawari-AHI%2C+Full+Disk%2C+Sensor+Geometry) shows the proper interpretation near the top of the page for similar satellite, Himawari.

## Additional information

I created a bug fixed code that shows a match to the Himawari example as shown below.

g16_8km_MetaData_sensorAzimuthAngle

### PR DESCRIPTION
**Pull request by gthompsnJCSDA**
## Description

Full description found in #1229.
In addition, the sensorViewAngle was an identical duplicate of sensorZenithAngle, so I removed this variable.

### Issue(s) addressed

Closes #1229 

## Acceptance Criteria (Definition of Done)

Review approvals

## Dependencies

None

## Impact

No impact to other repos expected.

### CODE REVIEW

**Comment by ericlingerfelt** [comment]
@gthompsnJCSDA, Does this PR affect https://github.com/JCSDA-internal/r2d2-ingest/issues/14? That is, will the files listed in r2d2-ingest@14 be modified?

**Review by ericlingerfelt — APPROVED** [review]

**Comment by gthompsnJCSDA** [comment]
> @gthompsnJCSDA, Does this PR affect [JCSDA-internal/r2d2-ingest#14](https://github.com/JCSDA-internal/r2d2-ingest/issues/14)? That is, will the files listed in r2d2-ingest@14 be modified?


Yes, all the files in my S3 bucket will need to be re-updated (again).  So please hold off on transfer of those files for a little bit.  I will put that PR into draft mode for now.  Probably tomorrow or this weekend, I will have all the files updated and then take 14 out of draft mode.  Does that work well for you?

File: src/goes/goes_latlon.py [hunk 1]
```diff
@@ -138,40 +139,43 @@ def _calc_scan_position(latitude):
...
-        sensor_azimuth_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(sensor_azimuth_angle)
-        sensor_view_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(sensor_view_angle)
-        sensor_zenith_angle = np.nan_to_num(sensor_zenith_angle, nan=-999)
-        sensor_azimuth_angle = np.nan_to_num(sensor_azimuth_angle, nan=-999)
-        sensor_view_angle = np.nan_to_num(sensor_view_angle, nan=-999)
+        sensor_zenith_angle = np.arcsin((h * np.sin(beta)) / np.sqrt(sqrt_comp)) / d2r
+        sensor_view_angle = np.arcsin(r_eq / h * np.sin(sensor_zenith_angle*d2r)) / d2r
+        x = np.sin(longitude_rad - lon_0_rad) * np.cos(latitude_rad)
+        y = np.cos(lat_0_rad)*np.sin(latitude_rad) - np.sin(lat_0_rad)*np.cos(latitude_rad)*np.cos(longitude_rad - lon_0_rad)
+        sensor_azimuth_angle = np.arctan2(x,y) / d2r + 180.0
```
Comment (PatNichols) [inline src/goes/goes_latlon.py:166]: @gthompsnJCSDA  Missing a whitespace here after the x. This is causing  a coding norms failure

Comment (PatNichols) [reply src/goes/goes_latlon.py]: I added it to kick off tests.

**Review by PatNichols — APPROVED** [review]
Looks good and thanks for finding this.

File: src/goes/goes_latlon.py (hunk 1 above)
Comment (gthompsnJCSDA) [reply src/goes/goes_latlon.py]: Thanks for fixing it for me!

File: src/goes/goes_latlon.py [hunk 2]
```diff
@@ -138,40 +139,43 @@ def _calc_scan_position(latitude):
...
-        sensor_azimuth_angle = np.nan_to_num(sensor_azimuth_angle, nan=-999)
-        sensor_view_angle = np.nan_to_num(sensor_view_angle, nan=-999)
+        sensor_zenith_angle = np.arcsin((h * np.sin(beta)) / np.sqrt(sqrt_comp)) / d2r
+        sensor_view_angle = np.arcsin(r_eq / h * np.sin(sensor_zenith_angle*d2r)) / d2r
+        x = np.sin(longitude_rad - lon_0_rad) * np.cos(latitude_rad)
+        y = np.cos(lat_0_rad)*np.sin(latitude_rad) - np.sin(lat_0_rad)*np.cos(latitude_rad)*np.cos(longitude_rad - lon_0_rad)
+        sensor_azimuth_angle = np.arctan2(x, y) / d2r + 180.0
         sensor_zenith_angle = np.where(sensor_zenith_angle > 80, 80, sensor_zenith_angle)
-        sensor_zenith_angle = np.where(sensor_zenith_angle < -80, -80, sensor_zenith_angle)
+        sensor_zenith_angle = np.where(sensor_zenith_angle < 0, 0, sensor_zenith_angle)
```
Comment (chengdang) [inline src/goes/goes_latlon.py:168]: @gthompsnJCSDA @BenjaminTJohnson Hello, Greg, could you please double check if the 80 degree threshold is removed or not? From this latest PR, it seems this 80-degree gets updated again.

**Comment by gthompsnJCSDA** [comment]
@chengdang   I think I've addressed this question already.  The `sensorZenithAngle` is already less than 80 degrees - see this figure below.  The sudden shift in values occur for the `solarZenithAngle` as I showed already in the figures of the open UFO issue.  Note the word difference to **solar** from _sensor_.


g16_8km_MetaData_sensorZenithAngle

**Comment by chengdang** [comment]
Thanks for sharing the figure! Here is what I think:
1. If the 80-degree threshold is not used anymore and we couldn't find this criteria in CRTM, can we consider remove this in ioda?  @BenjaminTJohnson 
2. Sensor zenith angles can be larger than 80 degree (for example, https://opus.nci.org.au/display/NDP/Himawari-AHI%2C+Full+Disk%2C+Sensor+Geometry), could you expand the colorbar of this plot to 90 degree? With the current colorbar, everything >=80 will be plot as red.
3. I can run some offline CRTM tests with both sensor zenith/solar zenith angles from 0-100, so we can see how big a leap is in radiance across the 81/90 degree line.

**Comment by gthompsnJCSDA** [comment]
Cheng,  Sorry, but I disagree completely that it is worth my time to mess around with the final 10-degrees of possibility with sensorZenithAngle, because I've shown really, really clearly what is happening at 81 solarZenithAngle and it would just waste more time to consider it.  Please study the two figures near the bottom of UFO 2753.  We have ruled out sensor angle as the cause of problem.  It's time to move past this discussion of sensorZenithAngle please.

**Comment by BenjaminTJohnson** [comment]
I agree with Greg regarding Sensor zenith angle, I generally recommend that
people ignore everything above 70 anyway for various reasons (including but
not limited to geometry).

Solar zenith angle, however, still needs attention.  I'm currently in the
airport, and will get back to this on Monday.
Clearly we want to accurately simulate solar zenith angles from zenith to
at least 120 degrees (?), to catch upper atmosphere solar effects.
However,
it appears that the transmittance model may stop at 85 degrees solar angle
in the transmittance coefficient training process.  I will talk to Patrick
about this.



Thanks,

Ben

**Comment by chengdang** [comment]
Sounds good! Then I guess the only thing left for this issue is to wait for Yingtao's updated cloud coeff table. Thanks.

**Pull request merged**